index.bs

```diff
@@ -480,10 +480,11 @@ that shares the same buffer as the input tensor. (In the case of reshape or sque
 the entire data is shared, while in the case of slice, a part of the input data is shared.)
 The implementation may use views, as above, for intermediate values.
 
-The {{MLGraphBuilder}}.{{MLGraphBuilder/build()}} method of the {{MLGraphBuilder}} interface is used to compile and optimize
-the computation graph used to compute one or more specified outputs. The key
-purpose of the compilation step is to enable optimizations that span two or
-more operations, such as operation or loop fusion.
+Before the execution, the computation graph that is used to compute one or more specified outputs needs to be compiled and optimized. The key purpose of the compilation step is to enable optimizations that span two or more operations, such as operation or loop fusion.
+
+There are multiple ways by which the graph may be compiled. The {{MLGraphBuilder}}.{{MLGraphBuilder/build()}} method represents a way the compilation of the graph is carried out immediately on the calling thread, which must also be a worker thread, either on a CPU or GPU device. The compilation produces the {{MLGraph}} that represents a compiled graph for optimal execution.
+
+The {{MLGraphBuilder}}.{{MLGraphBuilder/buildAsync()}} method represents a way the compilation of the graph is performed asynchronously either on a parallel timeline in a separate worker thread for the CPU execution or on a GPU timeline in a GPU command queue. This method returns immediately without blocking the calling thread while the actual compilation is offloaded to a different timeline. This type of compilation is appropriate when the responsiveness of the calling thread is critical to good user experience. The {{MLGraph}} will be returned at the time the compilation is successfully completed on the offloaded timeline at which time the calling thread is signaled.
```

Comment:
WDYT: 

buildAsync() compiles the graph in background without blocking the calling thread, and returns a {{Promise}} that resolves to {{MLGraph}}.